tests/framework/extensions/users/users.go

```diff
@@ -96,3 +98,14 @@ func RemoveProjectMember(rancherClient *rancher.Client, user *management.User) e
 	}
 	return rancherClient.Management.ProjectRoleTemplateBinding.Delete(&roleToDelete)
 }
+
+// AsUser accepts a user object, and then creates a token for said `user`. Then it instantiates and returns a Client using the token created.
+// This function uses the login action, and user must have a correct username and password combination.
+func AsUser(s suite.Suite, c *rancher.Client, user *management.User) (*rancher.Client, error) {
+	returnedToken, err := auth.Login(s, c, user)
```

Comment:
I think this would be better in` tests/framework/clients/rancher/client.go`, as it's returning a client instantiated of the `user` parameter.  `WithSession(session *session.Session)` does something similar, so I think it fits better there. Everything else is good, that is my last nitpick. Good job!

Reply:
Awesome thanks! Just had one question about the `project-member` test. Since the testnamespace is created with the testuser client it tries to delete the namespace on cleanup, but it doesn't have the correct rolebinding to be able to do so (and as far as I can tell, [only admin can delete namespaces](https://github.com/rancher/rancher/blob/d8ebe6252ec8e6b6dcd55ae3c592477842a43d93/pkg/data/management/role_data.go#L86-L88)). Should I do anything about this? I was able to remove one of two errors (as `project-members` cannot watch the namespace either), but do you think it would be more user friendly to pass the admin client that we wish to use for cleanup when we delete the namespace (or do you have something more straightforward in mind)?

Reply:
Interesting, I was under the impression, that a project-member (as long as they are a member of that specific project) were allowed to created and delete a namespace. Hmmm ... that seems wrong, I checked a test plan we had and that is the expected behavior

Reply:
I just verified that a user can create and delete namespaces with the `project-member` role, not sure why this is any different. I'll update when I have more info, but for now, these are the errors I receive:
```
session.go:47: error calling cleanup function: unknown
session.go:47: error calling cleanup function: namespaces "testnamespace" is forbidden: User "u-c2h7z" cannot delete resource "namespaces" in API group "" in the namespace "testnamespace": RBAC: clusterrole.rbac.authorization.k8s.io "p-hhg9w-namespaces-edit" not found
```
The `unknown` error comes from [here](https://github.com/rancher/rancher/blob/32fa450375fe52720ce2f44db1b243736929ea21/tests/framework/extensions/namespaces/namespaces.go#L70-L73), which is actually a 403 response like this:
```
{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"namespaces \"testnamespace\" is forbidden: User \"u-6cxpn\" cannot watch resource \"namespaces\" in API group \"\" at the cluster scope: RBAC: clusterrole.rbac.authorization.k8s.io \"p-gtskb-namespaces-edit\" not found","reason":"Forbidden","details":{"name":"testnamespace","kind":"namespaces"},"code":403}
```
Also, I may have to move the login action from `extensions/auth`, or move `AsUser` from `client`, as currently they are creating a cyclic package dependency. Do you have any strong preference one way or the other for how I should approach that?

Reply:
Sounds good Jake thanks for doing that. Move `Login` I think `AsUser` best fits in `client.go` Thank you!